sdk/storage/azblob/README.md

```diff
@@ -1,443 +1,276 @@
-```
+Blob storage offers three types of resources:
 
-Use Shared Key authentication as the credential parameter to authenticate the client:
+- The _storage account_
+- One or more _containers_ in a storage account
+- One ore more _blobs_ in a container
 
-```golang
-import "github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
+Instances of the `azblob.Client` type provide methods for manipulating containers and blobs within a storage account.
+The storage account is specified when the `azblob.Client` is constructed.
+Use the appropriate client constructor function for the authentication mechanism you wish to use.
 
-serviceURL := fmt.Sprintf("https://%s.blob.core.windows.net/", accountName)
+Learn more about options for authentication _(including Connection Strings, Shared Key, Shared Key Signatures, Active Directory, and anonymous public access)_ [in our examples.](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/examples_test.go)
 
-cred, err := service.NewSharedKeyCredential(accountName, accountKey)
-handleError(err)
+### Goroutine safety
+We guarantee that all client instance methods are goroutine-safe and independent of each other ([guideline](https://azure.github.io/azure-sdk/golang_introduction.html#thread-safety)). This ensures that the recommendation of reusing client instances is always safe, even across goroutines.
 
-serviceClient, err := service.NewClientWithSharedKey(serviceURL, cred, nil)
-handleError(err)
+### About blob metadata
+Blob metadata name/value pairs are valid HTTP headers and should adhere to all restrictions governing HTTP headers. Metadata names must be valid HTTP header names, may contain only ASCII characters, and should be treated as case-insensitive. Base64-encode or URL-encode metadata values containing non-ASCII characters.
 
-fmt.Println(serviceClient.URL())
-```
+### Additional concepts
+<!-- CLIENT COMMON BAR -->
+[Client options](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/policy#ClientOptions) |
+[Accessing the response](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime#WithCaptureResponse) |
+[Handling failures](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore#ResponseError) |
+[Logging](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/log)
+<!-- CLIENT COMMON BAR -->
 
-##### 2. Creating the client from a connection string
+## Examples
 
-You can use connection string, instead of providing the account URL and credential separately, for authentication as
-well. To do this, pass the connection string to the client's `NewServiceClientFromConnectionString` method. The
-connection string can be found in your storage account in the [Azure Portal][azure_portal_account_url] under the "Access
-Keys" section or with the following Azure CLI command:
+### Uploading a blob
 
-```bash
-az storage account show-connection-string -g my-resource-group -n mystorageaccount
-```
+```go
+const (
```

Comment:
These'll migrate into their own example files, rather than being inlined at some point, right?

Reply:
I think having bare-bones, inline examples are still useful as you can demonstrate the API without any ceremony (I stole this idea from the C# version that I copy/pasta'ed).  Full end-to-end samples should go in source files.